tests/test_completion.py

```diff
@@ -603,6 +607,36 @@ def test_text_edit_plaintext_with_multiple_lines_indented(self) -> Generator[Non
         # the "b" should be intended one level deeper
         self.assertEqual(self.read_file(), '\t\n\ta\n\t\tb')
 
+    def test_insert_insert_mode(self) -> 'Generator':
+        self.type('{{ title }}')
+        self.move_cursor(0, 5)  # Put the cursor inbetween 'i' and 't'
+        self.set_response("textDocument/completion", [{
+           'label': 'title',
+           'textEdit': {
+                'newText': 'title',
+                'insert': {'start': {'line': 0, 'character': 3}, 'end': {'line': 0, 'character': 5}},
+                'replace': {'start': {'line': 0, 'character': 3}, 'end': {'line': 0, 'character': 8}}
+            }
+        }])
+        yield from self.select_completion()
+        yield from self.await_message("textDocument/completion")
+        self.assertEqual(self.read_file(), '{{ titletle }}')
+
+    def test_replace_insert_mode(self) -> 'Generator':
+        self.type('{{ title }}')
+        self.move_cursor(0, 5)  # Put the cursor inbetween 'i' and 't'
+        self.set_response("textDocument/completion", [{
+           'label': 'title',
+           'textEdit': {
+                'newText': 'title',
+                'insert': {'start': {'line': 0, 'character': 3}, 'end': {'line': 0, 'character': 5}},
+                'replace': {'start': {'line': 0, 'character': 3}, 'end': {'line': 0, 'character': 8}}
+            }
+        }])
+        yield from self.shift_select_completion()  # commit the opposite insert mode
+        yield from self.await_message("textDocument/completion")
+        self.assertEqual(self.read_file(), '{{ title }}')
```

Comment:
Can you change `newText` to something else than the initial text to ensure that we don't pass if this just doesn't do anything?